pvl/parser.py

```diff
@@ -627,6 +627,8 @@ def parse_value(self, tokens: abc.Generator):
                 try:
                     value = p(tokens)
                     break
+                except LexerError:
+                    raise
                 except ValueError:
                     pass
```

Comment:
what happens here that we can afford to silently pass?

Reply:
This has to do with how the parsing logic works.  I'll admit this part is a little clunky, but it works.

You'll see that this try-block is inside a for loop and 'p' takes on different functions as we 'try out' different ways to parse the next set of tokens.  We try to parse the tokens as a set, and then as a sequence, and finally allow a user-specified hook to attempt to parse them.  Parsing 'success' results in a return to the 'value' variable, and clean break out of the for loop and on to the next thing.  Parsing failure results in a ValueError.  The 'pass' allows the code to try the next one in the list.  The reason for the change is that our custom LexerError is a subclass of ValueError.  So if we get a LexerError, that's a real problem and we need to get out, not just try the next thing.  If we run out of the list, there's a concluding else on the for statement that throws back into the lexer.

Reply:
this would be good to capture in a inline comment in that section